operator/builtin/input/file/config.go

```diff
@@ -151,10 +151,11 @@ func (c InputConfig) Build(context operator.BuildContext) ([]operator.Operator,
 		firstCheck:         true,
 		cancel:             func() {},
 		knownFiles:         make([]*Reader, 0, 10),
+		tailingReaders:     map[string][]*Reader{},
```

Comment:
What does the slice of Readers represent?

Reply:
If it is all previously rotations of the file, then do we really need to keep them? When we detect a rotation, can we just read the rest of the old file and then close it? Do we expect that rotated files are still written to? (My understanding is that the issue we have is that a file is written to immediately before being rotated.)

Reply:
good point. once file is rotated, it won't be written to. I should add a workflow to stop tailing them yea?